layers/meta-tegra-support/recipes-ota/tegra-bup-payload/tegra-bup-payload_%.bbappend

```diff
@@ -0,0 +1,3 @@
+RDEPENDS_${PN}_remove = "tegra-redundant-boot"
+RDEPENDS_${PN}_append = " tegra-boot-tools-updater"
+
```

Comment:
Here, too, this will affect all images that include a BUP payload. I'm generally OK with this, I guess. I had held off on doing this in the dunfell-based branches, but did this in gatesgarth and later. You should review the changes I made in the gatesgarth branch over the Xmas holiday to see if they might also be needed:
```
$ git log --reverse --pretty=oneline --since=25-dec-2020 --before=4-jan-2021 origin/gatesgarth
```

Reply:
> I had held off on doing this in the dunfell-based branches
Was there a reason for doing this?

Reply:
I just didn't want to run the risk of introducing new tools into the LTS branches.

Reply:
I agree with that. Is there any other solutions you might think of for this? 

I was wondering if we could include tegra-bootloader-update as part of sysinstall-upgrader-initramfs.

Reply:
Well, the sysinstall scripts chroot into the installed rootfs to run the bootloader update, so that the tools/config/payload are all for the new OS rather than the old OS under which the upgrader initramfs is built... and I think it's probably safer to keep it that way.

I'm OK with making these changes now, since automating the OTA reflashing process is a useful new feature.

Reply:
Alright

> You should review the changes I made in the gatesgarth branch over the Xmas holiday to see if they might also be needed

I'll take a look at this and get back.

Reply:
And FYI... @ichergui noticed that the changes I made earlier to the BUP payload contents cause failures when doing regular Mender-type updates (not reflashing) on the TX2.  The L4T `nv_update_engine` wasn't happy with the additions I made, so I've had to back them out and create a separate payload that can be used for a full initialization.

I've updated the test distros I mentioned on the wiki with the latest changes (including an update to the tegra-sysinstall scripts) to handle this.

Reply:
I have updated the branch to add whatever you have added on gatesgarth branch. I cross referenced the commits that are already added in `meta-mender-community`, updated with changes that were added for using `tegra-bootloader-update` tool.

Reply:
There's one more patch for the update scripts [here](https://github.com/madisongh/tegra-test-distro/commit/fdd6ee50c27f849b19b9a9bd24b9667d1c9abac8#diff-01e509973933159d027521681546fd193965ce207cb7e20cadbd19400440492a) that needs to be ported over, but otherwise this looks good.